frontend/src/employee-frontend/components/unit/unit-reservations/StaffAttendanceTable.tsx

```diff
@@ -119,6 +132,130 @@ export default React.memo(function StaffAttendanceTable({
     [extraAttendances]
   )
 
+  const [modalStaffAttendance, setModalStaffAttendance] = useState<
+    Result<StaffAttendanceResponse>
+  >(Loading.of())
+  const [modalAttendance, setModalAttendance] = useState<
+    Result<UnitAttendanceReservations>
+  >(Loading.of())
+  const loadModalStaffAttendance = useRestApi(
+    getStaffAttendances,
+    setModalStaffAttendance
+  )
+  const loadModalAttendance = useRestApi(
+    getUnitAttendanceReservations,
+    setModalAttendance
+  )
+
+  const modalOperationalDays = useMemo(
+    () => modalAttendance.getOrElse(undefined)?.operationalDays,
+    [modalAttendance]
+  )
+
+  const changeDate = useCallback(
+    async (getNearestDay: GetNearestDayFn, newStartOfWeek: LocalDate) => {
+      if (!detailsModal) return
+
+      const nearestNextDate =
+        (modalOperationalDays &&
+          getNearestDay(modalOperationalDays, detailsModal.date)) ??
+        getNearestDay(operationalDays, detailsModal.date)
+
+      if (!nearestNextDate) {
+        const nextWeekRange = new FiniteDateRange(
+          newStartOfWeek,
+          newStartOfWeek.addDays(6)
+        )
+
+        const [_, attendance] = await Promise.all([
+          loadModalStaffAttendance(unitId, nextWeekRange),
+          loadModalAttendance(unitId, nextWeekRange)
+        ])
+
+        const newOperationalDays =
+          attendance.getOrElse(undefined)?.operationalDays
+
+        if (!newOperationalDays) return
+
+        const newPreviousDay = getNearestDay(
+          newOperationalDays,
+          detailsModal.date
+        )
+
+        if (!newPreviousDay) return
+
+        setDetailsModal({
+          ...detailsModal,
+          date: newPreviousDay.date
+        })
+        return
+      }
+
+      setDetailsModal({
+        ...detailsModal,
+        date: nearestNextDate.date
+      })
+    },
+    [
+      detailsModal,
+      loadModalAttendance,
+      loadModalStaffAttendance,
+      modalOperationalDays,
+      operationalDays,
+      unitId
+    ]
+  )
+
+  const personCountSums = useMemo(
+    () =>
+      mapValues(
+        groupBy(
+          staffRows
+            .flatMap(({ attendances, employeeId }) =>
+              uniq(
```

Comment:
I'm not sure `uniq` works here as intended...every element coming from the flatMap below is going to be a different LocalDate instance, so uniq will never deduplicate anything.